src/Table/Table.tsx

```diff
@@ -83,48 +75,60 @@ const Actions = styled(Td)(({ theme }) => ({
   },
 }))
 
+const InlineActionMenu = styled(ActionMenu)`
+  display: inline-flex;
+`
+
 const EmptyView = styled(Td)(({ theme }) => ({
   color: theme.color.text.default,
   height: 50,
   lineHeight: "50px",
   textAlign: "center",
 }))
 
-const Table: React.SFC<TableProps> = ({
-  rows,
-  columns,
-  onRowClick,
-  rowActionName,
-  __experimentalRowActions,
-  ...props
-}) => {
+function Table<T>({ data, columns, onRowClick, rowActions, ...props }: TableProps<T>) {
+  const standardizedColumns: Array<Column<T>> =
+    Boolean(columns[0]) && typeof columns[0] === "string"
+      ? (columns as Array<Extract<keyof T, string>>).map(columnName => ({
+          heading: columnName,
+          cell: (datum: T) => datum[columnName],
+        }))
+      : (columns as Array<Column<T>>)
```

Comment:
Maybe a newline after this for readability? 👼 
```suggestion
      : (columns as Array<Column<T>>)
      
```